logconfig/config.go

```diff
@@ -18,69 +18,77 @@
 package logconfig
 
 import (
-	"bytes"
-	"text/template"
+	"os"
+	"strconv"
+	"strings"
+	"time"
 
-	log "github.com/cihub/seelog"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
 )
 
-const seewayLogXMLConfigTemplate = `
-<seelog minlevel="{{.LogLevel}}">
-	<outputs formatid="main">
-		<console/>
-		{{ if (ne .Filepath "") }}
-		<rollingfile
-			formatid="main"
-			filename="{{.Filepath}}"
-			maxrolls="7"
-			type="date"
-			datepattern="2006.01.02"
-		/>
-		{{ end }}
-	</outputs>
-	<formats>
-		<format id="main" format="%UTCDate(2006-01-02T15:04:05.999999999) [%Level] %Msg%n"/>
-	</formats>
-</seelog>
-`
-
-func buildXmlConfig(opts LogOptions) string {
-	tmpl := template.Must(template.New("seelogcfg").Parse(seewayLogXMLConfigTemplate))
-
-	var tpl bytes.Buffer
-	err := tmpl.Execute(&tpl, opts)
-	if err != nil {
-		panic(err)
-	}
-
-	return tpl.String()
-}
+// TODO remove this after log rotation is re-done on zerolog
+// leaving for the reference
+//const seewayLogXMLConfigTemplate = `
+//<seelog minlevel="{{.LogLevel}}">
+//	<outputs formatid="main">
+//		<console/>
+//		{{ if (ne .Filepath "") }}
+//		<rollingfile
+//			formatid="main"
+//			filename="{{.Filepath}}"
+//			maxrolls="7"
+//			type="date"
+//			datepattern="2006.01.02"
+//		/>
+//		{{ end }}
+//	</outputs>
+//	<formats>
+//		<format id="main" format="%UTCDate(2006-01-02T15:04:05.999999999) [%Level] %Msg%n"/>
+//	</formats>
+//</seelog>
+//`
 
 // BootstrapWith loads log package into the overall system
 func BootstrapWith(opts *LogOptions) {
 	if opts != nil {
 		CurrentLogOptions = *opts
 	}
-	newLogger, err := log.LoggerFromConfigAsString(buildXmlConfig(CurrentLogOptions))
-	if err != nil {
-		log.Warn("Error parsing log configuration", err)
-		return
-	}
-	err = log.UseLogger(newLogger)
-	if err != nil {
-		log.Warn("Error setting new logger for log", err)
-	}
-	log.Infof("Log level: %s", CurrentLogOptions.LogLevel)
+	configureZerolog(opts)
+	log.Info().Msgf("Log level: %s", CurrentLogOptions.LogLevel)
 	if CurrentLogOptions.Filepath != "" {
-		log.Infof("Log file path: %s", CurrentLogOptions.Filepath)
+		log.Info().Msg("Log file path: " + CurrentLogOptions.Filepath)
 	}
 }
 
 // Bootstrap loads log package into the overall system with debug defaults
 func Bootstrap() {
 	BootstrapWith(
 		&LogOptions{
-			LogLevel: log.DebugStr,
+			LogLevel: zerolog.DebugLevel,
 		},
 	)
 }
+
+func padRight(s string, length int) string {
+	if len(s) >= length {
+		return s
+	}
+	return s + strings.Repeat(" ", length-len(s))
+}
+
+func configureZerolog(opts *LogOptions) {
+	zerolog.TimeFieldFormat = time.RFC3339Nano
+	zerolog.CallerMarshalFunc = func(file string, line int) string {
+		pkg := "github.com/mysteriumnetwork/node"
```

Comment:
As pkg variable is not changing it could be allocated once

```go
const rootPkg = "github.com/mysteriumnetwork/node"
func configureZerolog() {
 // ...
}
```
```

Reply:
It’s easier to read when it’s colocated. Besides, it’s only called once.

Reply:
I think it can't be called once as you get caller name from different files there actual logging is done during runtime.

Reply:
You are right about this one, forgot that it's invoked every time the marshal is made. I'll extract it.